File: backend/src/main/java/com/campusforum/resource/domain/Resource.java
package com.campusforum.resource.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.campusforum.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("resources")
public class Resource extends BaseEntity {
    private Long uploaderId;
    private Long spaceId;
    private String fileName;
    private Long fileSize;
    private String fileType;
    private String fileMd5;
    private String storageKey;
    private String visibility;
    private String college;
    private String major;
    private String course;
    private String semester;
    private String tags;
    private Integer downloadCount;
    private Integer collectCount;
    private String version;
    private String description;
    private Integer status;
}

File: frontend/src/types/resource.ts
export interface UserBrief {
  id: number;
  nickname: string;
  avatarUrl: string;
}

export interface ResourceVO {
  id: number;
  uploaderId: number;
  uploader: UserBrief | null;
  spaceId: number | null;
  fileName: string;
  fileSize: number;
  fileType: string;
  visibility: string;
  college: string | null;
  major: string | null;
  course: string | null;
  semester: string | null;
  tags: string[];
  downloadCount: number;
  collectCount: number;
  version: string | null;
  description: string | null;
  createdAt: string;
}

export interface UploadResourceRequest {
  spaceId?: number;
  visibility?: string;
  college?: string;
  major?: string;
  course?: string;
  semester?: string;
  tags?: string[];
  description?: string;
}
